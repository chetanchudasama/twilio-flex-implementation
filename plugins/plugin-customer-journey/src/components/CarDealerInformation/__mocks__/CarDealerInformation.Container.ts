import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
