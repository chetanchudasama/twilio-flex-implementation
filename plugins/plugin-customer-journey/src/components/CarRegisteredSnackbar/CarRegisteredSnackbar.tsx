import React from "react";
import { CustomSnackbar } from "@common/components";

import { CarRegisteredSnackbarProps } from "./CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = ({
  is247Cars,
}) => {
  return (
    <>
      {is247Cars ? (
        <CustomSnackbar
          type="success"
          message="Great news! This car is on our car search with one of our approved dealers!"
        />
      ) : (
        <CustomSnackbar
          type="info"
          message="This car isn't in our car search."
        />
      )}
    </>
  );
};
