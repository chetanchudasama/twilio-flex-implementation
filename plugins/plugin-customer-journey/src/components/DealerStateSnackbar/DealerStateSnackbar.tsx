import React, { ReactFragment, useEffect, useMemo, useState } from "react";
import {
  Grid,
  SnackbarContent,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import { Shared } from "@common/components";
import WarningIcon from "@material-ui/icons/Warning";
import InfoIcon from "@material-ui/icons/Info";

import { styles } from "./DealerStateSnackbar.Styles";
import { DealerGuideDialog } from "../DealerGuideDialog/DealerGuideDialog";
import { DealerStateType } from "../../common/enum";
import { DealerStateSnackbarProps } from "./DealerStateSnackbar.Props";

type Props = WithStyles<typeof styles> & DealerStateSnackbarProps;

const DealerStateSnackbarContent: React.FC<Props> = ({
  declinedDate,
  dealerState,
  reasonForDeclined,
  classes,
}) => {
  const [message, setMessage] = useState<ReactFragment>();
  const [type, setType] = useState<"error" | "warning">("warning");
  const [openDealerGuide, setOpenDealerGuide] = useState(false);

  const IconType = useMemo(
    () => (type === "error" ? InfoIcon : WarningIcon),
    [type]
  );

  useEffect(() => {
    const declinedMessage = (
      <>
        <div>This dealer has been declined:</div>
        <ul className={classes.listType}>
          {declinedDate && reasonForDeclined && (
            <li>
              {`${Shared.getFormattedDate(
                declinedDate,
                "DD/MM/YY"
              )} - ${reasonForDeclined}`}
            </li>
          )}
        </ul>
        <br />
        <span className={classes.declinedMessageLink}>See below</span>
        {" for similar vehicle recommendations from approved dealers."}
      </>
    );

    const unknownMessage = (
      <>
        <div>
          This dealer is unknown to us. Please confirm the checks below:
        </div>
        <br />
        <ul className={classes.listType}>
          <li>Is the dealer VAT registered?</li>
          <li>Are they FCA Authorised with credit broking permissions?</li>
          <li>Do they have a forecourt and signing?</li>
          <li>Do they have a professional website?</li>
        </ul>
        <br />
        {"For more information, "}
        <button
          type="button"
          className={classes.link}
          onClick={() => {
            setOpenDealerGuide(true);
          }}
        >
          read our guide
        </button>
        .
      </>
    );

    const approvedForDCIMessage = (
      <>
        {
          "This dealer is approved for DCI only. For information on how to proceed, "
        }
        <button
          type="button"
          className={classes.link}
          onClick={() => {
            setOpenDealerGuide(true);
          }}
        >
          read this
        </button>
        .
      </>
    );

    switch (dealerState) {
      case DealerStateType.Declined:
        setMessage(declinedMessage);
        setType("error");
        break;
      case DealerStateType.Unknown:
        setMessage(unknownMessage);
        setType("warning");
        break;
      case DealerStateType.ApprovedForDCI:
        setMessage(approvedForDCIMessage);
        setType("warning");
        break;
      default:
        break;
    }
  }, [
    classes.declinedMessageLink,
    classes.link,
    classes.listType,
    dealerState,
    declinedDate,
    reasonForDeclined,
  ]);

  return (
    <>
      <Grid item xs={12} className={classes.root}>
        <SnackbarContent
          className={`${classes.snackbarContainer} ${classes[type]}`}
          message={
            <Grid className={classes.content}>
              <IconType className={`${classes.icon} ${classes.iconVariant}`} />
              <span>{message}</span>
            </Grid>
          }
        />
      </Grid>
      {openDealerGuide && (
        <DealerGuideDialog
          open={openDealerGuide}
          handleDialogClose={() => {
            setOpenDealerGuide(false);
          }}
          dealerState={dealerState}
        />
      )}
    </>
  );
};
export const DealerStateSnackbar = withStyles(styles)(
  DealerStateSnackbarContent
);
