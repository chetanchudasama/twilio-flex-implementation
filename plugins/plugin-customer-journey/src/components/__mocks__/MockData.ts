import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
