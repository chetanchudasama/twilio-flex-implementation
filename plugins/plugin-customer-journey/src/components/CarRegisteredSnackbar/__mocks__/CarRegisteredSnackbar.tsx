import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
