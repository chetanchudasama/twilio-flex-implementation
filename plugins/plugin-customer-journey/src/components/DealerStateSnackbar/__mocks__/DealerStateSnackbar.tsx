import React from "react";
import { DealerStateSnackbarProps } from "../DealerStateSnackbar.Props";

export const DealerStateSnackbar: React.FC<DealerStateSnackbarProps> = (
  props
) => {
  return <div className="dealer-state-snackbar">DealerStateSnackbar</div>;
};
