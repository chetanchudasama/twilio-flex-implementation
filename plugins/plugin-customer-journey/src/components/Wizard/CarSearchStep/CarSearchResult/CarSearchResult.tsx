import React, { useMemo } from "react";
import NavigateNextIcon from "@material-ui/icons/NavigateNext";
import NavigateBeforeIcon from "@material-ui/icons/NavigateBefore";
import { Pagination, Shared } from "@common/components";

import { Grid } from "@material-ui/core";
import { v4 } from "uuid";
import { CarRegisteredSnackbar } from "../../../CarRegisteredSnackbar/CarRegisteredSnackbar";
import { SortDropdown } from "../../SortDropdown/SortDropdown";
import {
  VehicleSortingOptions,
  DealerStateType,
} from "../../../../common/enum";
import { StateToProps, DispatchToProps } from "./CarSearchResult.Props";
import CarSearchResultDetailWrapper from "../CarSearchResultDetailWrapper/CarSearchResultDetailWrapper.Container";
import { CarSearchResultStyles } from "./CarSearchResult.Styles";
import { DealerStateSnackbar } from "../../../DealerStateSnackbar/DealerStateSnackbar";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface OwnProps {}

export type ComponentProps = StateToProps & DispatchToProps & OwnProps;

const CarSearchResult: React.FC<ComponentProps> = ({
  filters,
  setFilters,
  vehicles,
  totalVehicles,
  vrmFilters,
}) => {
  const sort = useMemo(
    () => filters.sort ?? VehicleSortingOptions.PriceDescending,
    [filters.sort]
  );

  const offset = useMemo(() => {
    return (
      ((filters.page || Shared.defaultPage) - 1) *
      (filters.pageSize ||
        Number(
          process.env.REACT_APP_CAR_SEARCH_PAGE_SIZE || Shared.defaultPageSize
        ))
    );
  }, [filters.page, filters.pageSize]);

  const onClick = (
    _event: React.MouseEvent<HTMLElement>,
    _newOffset: number,
    page: number
  ) =>
    setFilters({
      ...filters,
      page,
    });

  const onSortChange = (event: React.ChangeEvent<HTMLSelectElement>) =>
    setFilters({
      ...filters,
      page: Shared.defaultPage,
      sort: Number(event.target.value) ?? null,
    });

  const dealerInfo = useMemo(() => {
    // for specification mode - filters.dealerId indicates searching by a specific dealer
    // for vrm mode - vrmFilters.dealerId indicates searching by a specific dealer
    if (
      (filters.dealerId || vrmFilters.dealerId) &&
      vehicles.length > 0 &&
      vehicles[0].dealer.state &&
      vehicles[0].dealer.state !== DealerStateType.Approved
    ) {
      return vehicles[0].dealer;
    }
    return null;
  }, [filters.dealerId, vehicles, vrmFilters.dealerId]);

  const showCarRegisteredSnackbar = useMemo(() => {
    // show snackbar when search mode is vrm, distance value will always be null in vrm mode
    // don't show car found snackbar if searching by dealer, as the dealer snackbar will be shown instead
    return filters.distance === null && !dealerInfo;
  }, [filters, dealerInfo]);

  const vehiclesComponent = (
    <>
      <Grid item xs={12} className="result-sort-container">
        <Grid
          container
          direction="row"
          justify="space-between"
          alignItems="center"
        >
          <Grid item className="result-text">
            {`${Shared.getFormattedNumber(totalVehicles)} ${
              totalVehicles > 1 ? "results" : "result"
            }`}
          </Grid>
          <Grid item>
            <SortDropdown sort={sort} onSortChange={onSortChange} />
          </Grid>
        </Grid>
      </Grid>
      {showCarRegisteredSnackbar && (
        <Grid item xs={12} className="result-sort-container">
          <CarRegisteredSnackbar is247Cars={vehicles[0]?.is247Cars} />
        </Grid>
      )}
      {dealerInfo && (
        <DealerStateSnackbar
          dealerState={dealerInfo.state}
          declinedDate={dealerInfo.declinedDate}
          reasonForDeclined={dealerInfo.declinedReason ?? ""}
        />
      )}
      {vehicles.map((vehicle) => (
        <Grid item xs={12} key={v4()}>
          <CarSearchResultDetailWrapper vehicleDetail={vehicle} />
        </Grid>
      ))}
      <Grid item xs={12}>
        <Pagination
          limit={
            filters.pageSize ||
            Number(
              process.env.REACT_APP_CAR_SEARCH_PAGE_SIZE ||
                Shared.defaultPageSize
            )
          }
          offset={offset}
          total={totalVehicles}
          onClick={onClick}
          centerRipple
          otherPageColor="default"
          nextPageLabel={<NavigateNextIcon />}
          previousPageLabel={<NavigateBeforeIcon />}
          innerButtonCount={1}
          css=""
        />
      </Grid>
    </>
  );

  const noVehiclesComponent = (
    <Grid item xs={12} className="no-result">
      No cars found.
    </Grid>
  );

  return (
    <CarSearchResultStyles>
      {vehicles.length ? vehiclesComponent : noVehiclesComponent}
    </CarSearchResultStyles>
  );
};

export default CarSearchResult;
