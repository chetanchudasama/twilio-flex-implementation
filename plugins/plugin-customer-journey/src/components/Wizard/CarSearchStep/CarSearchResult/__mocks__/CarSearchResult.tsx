import React from "react";
import { ComponentProps } from "../CarSearchResult";

const CarSearchResult: React.FC<ComponentProps> = (props) => <div />;

export default CarSearchResult;
