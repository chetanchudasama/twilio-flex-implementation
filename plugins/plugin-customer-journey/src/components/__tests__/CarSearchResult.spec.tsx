import React from "react";
import { create } from "react-test-renderer";
import flushPromises from "flush-promises";
import { Manager } from "@twilio/flex-ui";
import { mount } from "enzyme";
import { CarRegisteredSnackbar } from "../CarRegisteredSnackbar/CarRegisteredSnackbar";
import { DealerStateType } from "../../common/enum";
import { DealerStateSnackbar } from "../DealerStateSnackbar/DealerStateSnackbar";
import CarSearchResult, {
  ComponentProps,
} from "../Wizard/CarSearchStep/CarSearchResult/CarSearchResult";
import { VehicleSummaryDetailModel } from "../../models/VehicleSummaryDetailModel";
import { VehicleSearchRequestModel } from "../../models/VehicleSearchRequestModel";
import CarSearchResultDetailWrapperContainer from "../Wizard/CarSearchStep/CarSearchResultDetailWrapper/CarSearchResultDetailWrapper.Container";
import { flexManagerMock } from "../__mocks__/MockData";
import { VehicleVRMSearchRequestModel } from "../../models/VehicleVRMSearchRequestModel";

jest.mock(
  "../Wizard/CarSearchStep/CarSearchResultDetailWrapper/CarSearchResultDetailWrapper.Container"
);
jest.mock(
  "../Wizard/CarSearchStep/CarSearchResultDetailWrapper/CarSearchResultDetailWrapper"
);
jest.mock("../CarRegisteredSnackbar/CarRegisteredSnackbar");

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
  };
});

Manager.getInstance = flexManagerMock;

const defaultProps: ComponentProps = {
  vehicles: [],
  filters: new VehicleSearchRequestModel(),
  totalVehicles: 0,
  setFilters: jest.fn(() => {}),
  vrmFilters: new VehicleVRMSearchRequestModel(),
};

let props: ComponentProps = defaultProps;

Manager.getInstance = flexManagerMock;

describe("CarSearchResult", () => {
  beforeEach(() => {
    props = { ...defaultProps };
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it("renders correctly, if no vehicle found", async () => {
    const wrapper = create(<CarSearchResult {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, if vehicle found in search list", async () => {
    props.totalVehicles = 1;
    props.vehicles = [new VehicleSummaryDetailModel()];

    const wrapper = create(<CarSearchResult {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render vehicle details wrapper component if vehicles present", async () => {
    props.totalVehicles = 1;
    props.vehicles = [new VehicleSummaryDetailModel()];

    const wrapper = mount(<CarSearchResult {...props} />);
    await flushPromises();
    wrapper.update();

    expect(
      wrapper.find(CarSearchResultDetailWrapperContainer).exists()
    ).toBeTruthy();
  });

  it("should not render vehicle details wrapper component if vehicles are not present", async () => {
    props.totalVehicles = 0;
    props.vehicles = [];

    const wrapper = mount(<CarSearchResult {...props} />);
    await flushPromises();
    wrapper.update();

    expect(
      wrapper.find(CarSearchResultDetailWrapperContainer).exists()
    ).toBeFalsy();
  });

  it("should show no cars found message if no vehicles found", async () => {
    props.totalVehicles = 0;
    props.vehicles = [];

    const wrapper = mount(<CarSearchResult {...props} />);
    await flushPromises();
    wrapper.update();

    expect(wrapper.html()).toContain("No cars found.");
  });

  it("should show number of vehicles found in the search", async () => {
    props.totalVehicles = 1;
    props.vehicles = [new VehicleSummaryDetailModel()];

    const wrapper = mount(<CarSearchResult {...props} />);
    await flushPromises();
    wrapper.update();

    expect(wrapper.html()).toContain("1 result");
  });

  it("when value of distance is null in filters, should render CarRegisteredSnackbar in UI", async () => {
    props.filters.distance = null;
    props.totalVehicles = 1;
    props.vehicles = [new VehicleSummaryDetailModel()];
    const wrapper = mount(<CarSearchResult {...props} />);

    expect(wrapper.find(CarRegisteredSnackbar).exists()).toBeTruthy();
  });

  it("should not render DealerStateSnackbar component, if dealerId is null", async () => {
    props.totalVehicles = 1;
    props.vehicles = [new VehicleSummaryDetailModel()];

    const wrapper = mount(<CarSearchResult {...props} />);
    await flushPromises();
    wrapper.update();

    expect(wrapper.find(DealerStateSnackbar).exists()).toBeFalsy();
  });

  it("should not render DealerStateSnackbar component, if dealer state is approved", async () => {
    props.totalVehicles = 1;
    props.filters.dealerId = 2;
    props.vehicles = [new VehicleSummaryDetailModel()];
    props.vehicles[0].dealer.state = DealerStateType.Approved;

    const wrapper = mount(<CarSearchResult {...props} />);
    await flushPromises();
    wrapper.update();

    expect(wrapper.find(DealerStateSnackbar).exists()).toBeFalsy();
  });

  it("should render DealerStateSnackbar component, if dealerId is not null and dealer state is not approved", async () => {
    props.totalVehicles = 1;
    props.filters.dealerId = 2;
    props.vehicles = [new VehicleSummaryDetailModel()];
    props.vehicles[0].dealer.state = DealerStateType.Declined;

    const wrapper = mount(<CarSearchResult {...props} />);
    await flushPromises();
    wrapper.update();

    expect(wrapper.find(DealerStateSnackbar).exists()).toBeTruthy();
  });
});
