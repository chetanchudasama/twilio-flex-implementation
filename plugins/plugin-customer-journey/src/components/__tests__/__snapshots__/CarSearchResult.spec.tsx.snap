// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarSearchResult renders correctly, if no vehicle found 1`] = `
<div
  className="css-1ma0a9h"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 no-result"
  >
    No cars found.
  </div>
</div>
`;

exports[`CarSearchResult renders correctly, if vehicle found in search list 1`] = `
<div
  className="css-1ma0a9h"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 result-sort-container"
  >
    <div
      className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
    >
      <div
        className="MuiGrid-item-2 result-text"
      >
        1 result
      </div>
      <div
        className="MuiGrid-item-2"
      >
        <div
          className="MuiFormControl-root-103 MuiFormControl-marginDense-105 MuiFormControl-fullWidth-106 CustomSelectContent-formControl-98"
        >
          <div
            className="MuiInputBase-root-127 MuiOutlinedInput-root-114 MuiInputBase-formControl-128 MuiInputBase-marginDense-134 CustomSelectContent-select-100 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-144 MuiOutlinedInput-notchedOutline-121"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-145"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-107"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-108 MuiSelect-selectMenu-111 MuiInputBase-input-137 MuiOutlinedInput-input-122 MuiInputBase-inputMarginDense-138 MuiOutlinedInput-inputMarginDense-123"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                Sort by: Price (High to Low)
              </div>
              <input
                type="hidden"
                value={2}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-146 MuiSelect-icon-113"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 result-sort-container"
  >
    <div
      className="car-registered-snackbar"
    >
      CarRegisteredSnackbar
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div />
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiFlatPagination-root-166 WithStyles-Pagination--root-159"
      css=""
    >
      <button
        className="MuiButtonBase-root-240 MuiButtonBase-disabled-241 MuiButton-root-214 MuiFlatPageButton-root-190 MuiFlatPageButton-rootEnd-193 MuiFlatPagination-rootEnd-169 WithStyles-Pagination--rootEnd-162 MuiButton-text-216 MuiFlatPageButton-text-196 MuiFlatPagination-text-172 MuiButton-flat-219 MuiButton-disabled-234 MuiFlatPageButton-disabled-202 MuiFlatPagination-disabled-178"
        disabled={true}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-215 MuiFlatPageButton-label-195 MuiFlatPagination-label-171"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-146"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-240 MuiButton-root-214 MuiFlatPageButton-root-190 MuiFlatPageButton-rootCurrent-191 MuiFlatPagination-rootCurrent-167 WithStyles-Pagination--rootCurrent-161 MuiButton-text-216 MuiFlatPageButton-text-196 MuiFlatPagination-text-172 MuiButton-textSecondary-218 MuiFlatPageButton-textSecondary-198 MuiFlatPagination-textSecondary-174 WithStyles-Pagination--textSecondary-165 MuiButton-flat-219 MuiButton-flatSecondary-221"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-215 MuiFlatPageButton-label-195 MuiFlatPagination-label-171"
        >
          1
        </span>
      </button>
      <button
        className="MuiButtonBase-root-240 MuiButtonBase-disabled-241 MuiButton-root-214 MuiFlatPageButton-root-190 MuiFlatPageButton-rootEnd-193 MuiFlatPagination-rootEnd-169 WithStyles-Pagination--rootEnd-162 MuiButton-text-216 MuiFlatPageButton-text-196 MuiFlatPagination-text-172 MuiButton-flat-219 MuiButton-disabled-234 MuiFlatPageButton-disabled-202 MuiFlatPagination-disabled-178"
        disabled={true}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-215 MuiFlatPageButton-label-195 MuiFlatPagination-label-171"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-146"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
